feedback_prompt_learning/search_algo/mcts.py

```
@@ -109,13 +109,13 @@ def __init__(
         self,
         initial_prompt: str,
         world_model: WorldModel,
-        llm_action: ChatOpenAI = hydra_instantiate(cfg.optimizer.llm.action),
-        llm_critic: ChatOpenAI = hydra_instantiate(cfg.optimizer.llm.critic),
-        num_iterations: int = cfg.optimizer.num_iterations,
-        exploration_constant: float = cfg.optimizer.exploration_constant,
-        max_depth: int = cfg.optimizer.max_depth,
-        expand_width: int = cfg.optimizer.expand_width,  # Number of gradient analyses per expansion
-        num_samples: int = cfg.optimizer.num_samples,   # Number of prompts generated per gradient
+        llm_action: ChatOpenAI = hydra_instantiate(config._cfg.optimizer.llm.action),
```

Ruff (0.14.8)

Do not perform function call `hydra_instantiate` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)